Sprint-2/5-stretch-extend/format-time.js

```diff
@@ -4,22 +4,59 @@
 
 function formatAs12HourClock(time) {
   const hours = Number(time.slice(0, 2));
+  const minutes = time.slice(3,5); // extra minutes
+  // for midnight 
+  if (hours === 24 || hours === 0){
+    return `12:${minutes} am`;
+  }
+
   if (hours > 12) {
-    return `${hours - 12}:00 pm`;
+    return `${pad(hours- 12)}:${minutes} pm`;
+  }
+  
+  if (hours === 12){
+    return `${pad(hours)}:${minutes} pm`;
   }
-  return `${time} am`;
-}
 
+  return `${pad(hours)}:${minutes} am`;
+}
+// to pad the hours to always show two digits
+function pad(num) {
+  return num.toString().padStart(2,"0");
+}
+//test 1 (8:00) am
 const currentOutput = formatAs12HourClock("08:00");
 const targetOutput = "08:00 am";
 console.assert(
   currentOutput === targetOutput,
   `current output: ${currentOutput}, target output: ${targetOutput}`
 );
 
+// test 2 (11:00) pm
 const currentOutput2 = formatAs12HourClock("23:00");
 const targetOutput2 = "11:00 pm";
 console.assert(
   currentOutput2 === targetOutput2,
   `current output: ${currentOutput2}, target output: ${targetOutput2}`
 );
+
+//test 3 (midnight)
+const currentOutput3 = formatAs12HourClock("00:00");
+const targetOutput3 = "12:00 am";
+console.assert(
+  currentOutput3 === targetOutput3,
+  `current output: ${currentOutput3}, target output: ${targetOutput3}`
+);
+
+//test 4 noon
+const currentOutput4 = formatAs12HourClock("12:00");
+const targetOutput4 = "12:00 pm";
+console.assert(
+  currentOutput4 === targetOutput4,
+  `current output: ${currentOutput4}, target output: ${targetOutput4}`
+);
+
+
```

Comment:
Nice! These are good cases. Did you try writing the code incorrectly to see the test fail? It's useful to know what this looks like - sometimes it's good to know that tests fail when we change code. It helps us to guarantee that our code is functioning as we have said it should do.